README.md

```diff
@@ -1,102 +1,189 @@
+    - `pathPattern` / `excludedPathPattern` : Regular expressions that will be tested on the node path.
+    - `workspace` : `live` or `default`, only request on the specified workspace will match.
+    - `nodeType` / `excludedNodeType` : Only request on nodes of these type will match.
+    - `withPermission` : Only request on nodes, where the user has this permission, will match.
+
+    ```yaml
+       grants:
+         - node:
+             pathPattern: /,/sites(/.*)?,/modules(/.*)?,/mounts(/.*)?
+             excludedPathPattern: /sites/[^/]+/users(/.*)?
+    ```
+
+### Auto-apply rules
+
+Scope can be automatically applied to requests based on an origin. It's checked against the `Origin` and `Referer` headers. 
+`hosted` or `same` mean that the rule will match if the request is coming from the same server.
+
+```yaml
+   auto_apply:
+     - origin: hosted
+     - origin: http://www.mysite.com
 ```
 
-Or a combination of all criterias - if all conditions must match (including the permission), access will be granted
-```    
-permission.digitallPostsInLive.pathPattern=/sites/digitall/.*
-permission.digitallPostsInLive.nodeType=jnt:post,jnt:message
-permission.digitallPostsInLive.workspace=live
-permission.digitallPostsInLive.permission=jcr:write
-```
+It's also possible to always apply the scope, whatever the request is. It can be used to have API that will always be granted.
 
-This rule will grant full access for all users claiming one of the scopes scope1, scope2 or scope3
+```yaml
+   auto_apply:
+     - always: true
 ```
-permission.jwtAccess.scope=scope1,scope2,scope3
+
+### User constraints
+
+Some scopes are only usable by specific users. You can set which permission a user should have on a node :
+
+```yaml
+   constraints:
+     - user_permission: manageModules
+       path: /sites
+       workspace: live
 ```
 
-This rule will require myPermission for all graphql call on /sites/secure
+Or simply restrict the scope to privileged users :
+
+```yaml
+   constraints:
+     - privileged_user: true
 ```
-permission.secure.api=graphql
-permission.secure.pathPattern=/sites/secure/.*
-permission.secure.requiredPermission=myPermission
+
+The scope will be available only to users who fulfill the constraints. It will never be applied for other users.
+
+### Configuration profiles 
+
+The user can choose a predefined security profile by setting a value in `security.profile`, in `org.jahia.modules.api.security.cfg` file
+
+- "default" profile is recommended one. It will not allow any API call from external origin, and from non-privileged users.
+- "compat" profile is more open and is compatible with the previous security-filter implementation. Most graphql/rest calls are allowed for any user
+- "open" profile allows every call.
+
+It's also possible to not use any profile (everything will be denied by default) - you will have to fully provide your own configuration. Without any configuration Jahia GUI will not be work.
```

Comment:
I'm not able to verify this, if I'm previously in open, then specify a profile with an incorrect name or empty, I can still perform graphql calls and access the administration.

I get this in the logs:
```
jahia               | 2021-07-21 12:16:27,280: INFO  [fileinstall] - Updating configuration from org.jahia.modules.api.security.cfg
jahia               | 2021-07-21 12:16:27,282: ERROR [PermissionServiceImpl] - Invalid security-filter profile :
jahia               | 2021-07-21 12:16:27,333: INFO  [JahiaDatabaseJournal] - Creating revision: 1553
jahia               | 2021-07-21 12:16:27,342: INFO  [AbstractBundlePersistenceManager] - cachename=versionBundleCache[ConcurrentCache@18df2e6a], elements=7924, usedmemorykb=10714, maxmemorykb=24576, access=954532, miss=7924
jahia               | 2021-07-21 12:16:27,345: INFO  [audit] - [1509] 1553  system @default:/module-management/configs (23682)
jahia               | 2021-07-21 12:16:27,359: INFO  [JahiaDatabaseJournal] - Creating revision: 1554
jahia               | 2021-07-21 12:16:27,364: INFO  [audit] - [1510] 1554  system @default:/module-management/configs (20940)
jahia               | 2021-07-21 12:16:27,370: INFO  [RulesListener] - Rules executed for default [deleted /module-management/configs/org.jahia.modules.api.security.cfg/jcr:content, deleted /module-management/configs/org.jahia.modules.api.security.cfg, /module-management/configs/org.jahia.modules.api.security.cfg/jcr:content/jcr:data] ... and 2 other nodes in 3ms
jahia-browsing-b    | 2021-07-21 12:16:27,382: ERROR [PermissionServiceImpl] - Invalid security-filter profile :
jahia-browsing-a    | 2021-07-21 12:16:27,383: ERROR [PermissionServiceImpl] - Invalid security-filter profile :
jahia-browsing-a    | 2021-07-21 12:16:29,926: INFO  [ClusterNode] - Processing revision: 1553
jahia-browsing-a    | 2021-07-21 12:16:29,927: INFO  [audit] - [1509] 1553  system @default:/module-management/configs
jahia-browsing-a    | 2021-07-21 12:16:29,936: INFO  [ClusterNode] - Processing revision: 1554
jahia-browsing-a    | 2021-07-21 12:16:29,936: INFO  [audit] - [1510] 1554  system @default:/module-management/configs
jahia-browsing-b    | 2021-07-21 12:16:32,195: INFO  [ClusterNode] - Processing revision: 1553
jahia-browsing-b    | 2021-07-21 12:16:32,195: INFO  [audit] - [1509] 1553  system @default:/module-management/configs
jahia-browsing-b    | 2021-07-21 12:16:32,196: INFO  [AbstractBundlePersistenceManager] - cachename=defaultBundleCache[ConcurrentCache@3ad24842], elements=8404, usedmemorykb=8331, maxmemorykb=98304, access=1163701, miss=12877
jahia-browsing-b    | 2021-07-21 12:16:32,198: INFO  [AbstractBundlePersistenceManager] - cachename=versionBundleCache[ConcurrentCache@a4314f7], elements=6070, usedmemorykb=9581, maxmemorykb=24576, access=1350772, miss=11766
jahia-browsing-b    | 2021-07-21 12:16:32,205: INFO  [ClusterNode] - Processing revision: 1554
jahia-browsing-b    | 2021-07-21 12:16:32,205: INFO  [audit] - [1510] 1554  system @default:/module-management/configs
```

Reply:
Indeed, fixed now